swiperefresh/src/main/java/com/google/accompanist/swiperefresh/SwipeRefreshIndicator.kt

```diff
@@ -14,107 +14,125 @@
+import androidx.compose.foundation.shape.CornerBasedShape
+import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.contentColorFor
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.scale
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.graphics.graphicsLayer
 import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.util.lerp
-
-private const val IndicatorMaxAlpha = 1f
-private const val IndicatorMinAlpha = 0f
-
-private val AlphaEasing: Easing = LinearOutSlowInEasing
-
-private const val MAX_PROGRESS_ANGLE = .8f
+import kotlin.math.min
 
 @Composable
 fun SwipeRefreshIndicator(
     isRefreshing: Boolean,
-    currentOffset: Float,
-    isSwipeInProgress: Boolean,
-    backgroundColor: Color = MaterialTheme.colors.surface,
-    elevation: Dp = 6.dp,
+    offset: Float,
+    scale: Boolean = false,
+    arrowEnabled: Boolean = true,
+    color: Color = MaterialTheme.colors.surface,
+    contentColor: Color = contentColorFor(color),
+    shape: CornerBasedShape = CircleShape,
+    size: IndicatorSize = IndicatorSize.DEFAULT,
+    elevation: Dp = 4.dp
 ) {
-    // This might look like we've just re-implemented Surface, and that's because we have.
-    val elevationPx = with(LocalDensity.current) { elevation.toPx() }
-    val elevationOverlay = LocalElevationOverlay.current
-    val absoluteElevation = LocalAbsoluteElevation.current + elevation
-    val bgColor = if (backgroundColor == MaterialTheme.colors.surface && elevationOverlay != null) {
-        elevationOverlay.apply(backgroundColor, absoluteElevation)
-    } else backgroundColor
-
-    val size = 40.dp
-    val sizePx = with(LocalDensity.current) { size.toPx() }
-
-    val modifier = Modifier
-        .size(size)
-        .graphicsLayer {
-            shadowElevation = elevationPx
-            shape = CircleShape
-            alpha = when {
-                // We animate the indicator in as the indicator is dragged in
-                currentOffset >= 0.5 -> lerp(
-                    start = IndicatorMinAlpha,
-                    stop = IndicatorMaxAlpha,
-                    fraction = AlphaEasing.transform((currentOffset / sizePx).coerceAtMost(1f))
-                )
-                else -> 0f
-            }
-        }
-        .background(color = bgColor, shape = CircleShape)
-        .clip(CircleShape)
-
-    Box(modifier) {
-        val ringPainter = remember {
-            ProgressRingPainter()
+    val animatedElevation by animateDpAsState(
```

Comment:
Why `animateDpAsState()` when this is just using `snap()`? This could just be an `if` (I think)

Reply:
Whoops, yeah agree